File: Microsoft Office Developer Documentation Team/Outlook-2010-Outlook-Social-Connector-OSC-3977/[C#]-Outlook 2010 Outlook Social Connector (OSC) Provider Templates/C#/OSCProvider_CS/OSCPerson.cs
using System;
using System.Diagnostics;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using OutlookSocialProvider;

namespace OSCProvider_CS
{
    class OSCPerson : OutlookSocialProvider.ISocialPerson 
    {
        #region ISocialPerson Members

        public string GetActivities(DateTime startTime)
        {
            //To-Do: Implement GetActivities
            return string.Empty;
        }

        public string GetDetails()
        {
            //To-Do: Implement GetActivities
            return string.Empty;
        }

        public string GetFriendsAndColleagues()
        {
            //To-Do: Implement GetFriendAndColleagues
            return string.Empty;
        }

        public string[] GetFriendsAndColleaguesIDs()
        {
            //To-Do: Implement GetFriendAndColleaguesIDs
            string[] result = { "" };
            return result;

        }

        public byte[] GetPicture()
        {
            //To-Do: Implement GetPicture
            return null;
        }

        public string GetProfileUrl()
        {
            //To-Do: Implement GetProfileUrl
            return string.Empty;
        }

        public string GetStatus()
        {
            //Not supported in OSC version 1.0 and version 1.1
            return string.Empty;
        }

        #endregion
    }
}

File: Microsoft Office Developer Documentation Team/Outlook-2010-Outlook-Social-Connector-OSC-3977/[VB]-Outlook 2010 Outlook Social Connector (OSC) Provider Templates/VB/OSCProvider_VB/OSCPerson.vb
Imports System
Imports System.Data
Imports System.Xml
Imports System.Xml.Serialization
Imports System.Diagnostics
Imports System.Drawing
Imports System.Text
Imports OutlookSocialProvider

Namespace OSCProvider_VB
    Public Class OSCPerson
        Implements OutlookSocialProvider.ISocialPerson

#Region "ISocialPerson Members"
        Public Function GetActivities(ByVal startTime As Date) As String Implements OutlookSocialProvider.ISocialPerson.GetActivities
            'To-Do: Implement GetActivities
            Return String.Empty
        End Function

        Public Function GetDetails() As String Implements OutlookSocialProvider.ISocialPerson.GetDetails
            'To-Do: Implement GetDetails
            Return String.Empty
        End Function

        Public Function GetFriendsAndColleagues() As String Implements OutlookSocialProvider.ISocialPerson.GetFriendsAndColleagues
            'To-Do: Implement GetFriendsAndColleagues
            Return String.Empty
        End Function

        Public Function GetFriendsAndColleaguesIDs() As String() Implements OutlookSocialProvider.ISocialPerson.GetFriendsAndColleaguesIDs
            'To-Do: Implement GetFriendAndColleaguesIDs
            Dim result As String() = {""}
            Return result
        End Function

        Public Function GetPicture() As Byte() Implements OutlookSocialProvider.ISocialPerson.GetPicture
            'To-Do: Implement GetPicture
            Return (HelperMethods.GetProviderJpeg())
        End Function

        Public Function GetStatus() As String Implements OutlookSocialProvider.ISocialPerson.GetStatus
            'Not supported in OSC version 1.0 and version 1.1
            Return String.Empty
        End Function
#End Region
    End Class
End Namespace

